Automattic/wp-calypso
# Plans 2023: Implement storage add on dropdown

Related to https://github.com/Automattic/martech/issues/1907

## Proposed Changes

* User should be able to select additional storage for Business and Ecommerce plans:
  * Hide dropdown behind feature flag because we're waiting for storage caps to actually be lowered
  * Render storage selection dropdown
  * Ensure that dropdown is only displayed for Business / Ecommerce yearly plans

* What will be implemented in follow-up PRs:
  * Dropdown Functionality ( https://github.com/Automattic/wp-calypso/pull/79104 )
    * Addon is added to the plans checkout cart automatically
    * Plans prices in the UI update accordingly
    * Storage values in the plans comparison grid update accordingly
  * Add tracking event when user interacts with storage selection dropdown

## GIF

### Onboarding
![2023-07-06 16 29 05](https://github.com/Automattic/wp-calypso/assets/5414230/eda51647-5f13-43cc-bb07-3999c54f1925)

### /plans
![2023-07-06 19 19 27](https://github.com/Automattic/wp-calypso/assets/5414230/5e4cd7d9-015f-494f-8b40-1b3fd370b9da)

## Testing Instructions



* Check out this branch
*  yarn && yarn start
* Navigate to any onboarding flow ( Ex. calypso.localhost:3000/start ) and complete steps until at plans pricing page
* Verify that the dropdown renders and different storage options can be selected
* Verify that the dropdown renders only for business and ecommerce plans for yearly plans
* Navigate to the /plans page and repeat the same verifications above
* Verify that the feature flag is working by either:
  * Navigating to /start/plans/?flags=-plans/upgradeable-storage
  * commenting out `"plans/upgradeable-storage"` in `config/development.json` and restarting the dev environment

## Pre-merge Checklist
